### PR TITLE
Get validation env

### LINKED ISSUE #7: capture validation environment
Due to performance differences between OS, R version and package versions, it is helpful to capture the validation environment. This should include:
- OS
- R Version
- package versions for any listed in Description "Depends" or "Imports".

### PR DESCRIPTION
**Pull request by mariev**
blocked by #8 
closes #7 

**Change Summary**
`vt_scrape_val_env`, a new function to capture the package validation environment via a combo of scraping the DESCRIPTION file + looking at current workspace.

*Algorithm*

1. Captures OS, R, and `sessionInfo()["otherPkgs"]` names & versions
1. Compares (1) to info in DESCRIPTION Depends/Imports/Suggests of the package being validated. Assumes: `here::here()` is package location.
1. Classifies into following groups:
    - system: OS/R
    - package_req: listed in Depends/Imports fields
    - extended_req: listed in Suggests fields
    - session: not listed but active e.g. in validation report that has not passed `devtools::check()`

### CODE REVIEW

File: R/scrape_val_env.R [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#'         \item session - packages in current workspace not captured via package_req/extended_req
+#'     }
+#' \item \code{detail} OS or version details
+#'
+#' }
+#' @importFrom here here
+#' @importFrom utils packageVersion sessionInfo
+#' @importFrom desc desc
+#' @export
+vt_scrape_val_env <- function(pkg = here()){
```
Comment (thebioengineer) [inline R/scrape_val_env.R:24]: would it make more sense to use `pkg = "."`? this is in alignment with devtools practices

**Review by thebioengineer — APPROVED** [review]
Looks good, code is clear and concise. 

One thought would be to add more extensive testing - ie making a tempdir with a sample description file. But I don't think that should block the PR

**Comment by mariev** [comment]
@thebioengineer commit `e5da851` should address your questions

File: R/scrape_val_env.R (hunk 1 above)
Comment (mariev) [reply R/scrape_val_env.R]: see `e5da851`

**Review by thebioengineer — APPROVED** [review]
Thanks for making those changes!

**Pull request merged**